dataprofiler/profilers/profile_builder.py

```diff
@@ -683,3 +687,127 @@ def tqdm(l):
             
         return profile
 
+    def _delete_data_labelers(self):
+        """
+        Helper method for deleting all data labelers before saving to disk.
+        """
+        # Delete data labeler for profiler
+        data_labeler_options = self.options.structured_options.data_labeler
+        if data_labeler_options.is_enabled \
+                and data_labeler_options.data_labeler_object is not None:
+                    data_labeler_options.data_labeler_object = None
```

Comment:
I know I had mentioned something like this, but does it work if we leave the data_labeler as well?

Also I don't believe this function in the current state actually deletes the labeler, you'd have to actually call the `del data_labaler` for it to get cleaned up.

Reply:
we shouldn't delete `data_labeler_object ` when saving the profile, as we might need it after saving.

Reply:
That's a fair observation - we should definitely avoid deleting!